router/group.py

```diff
@@ -0,0 +1,17 @@
+from fastapi import APIRouter, HTTPException
+import requests
+from settings import settings
+
+router = APIRouter(prefix="/group", tags=["Group"])
+group_db_url = settings.db_url + "group/"
+
+@router.post("/get-group-data")
+def get_group_data(group_id: str):
```

Comment:
request type is POST but method name is get?

Reply:
Also, if ultimate goal is to GET, why not keep the `group_id` in the main URL itself, and not query params?
Because one thing is that as this request is a POST request, you'll have to send group_id as a POST payload. It won't be coming in the URL itself.

Usually in GET requests, if there is only one type of value that needs to be passed along with the request, it can directly be embedded in the link.
Like the quiz id in quiz links - `https://quiz.avantifellows.org/quiz/627cec9817fda098baa3ed93?userId=1&apiKey=6qOO8UdF1EGxLgzwIbQN`

Reply:
I think for this also I can change to GET. Might help in speed also.